Integration - Google Maps (#32)
FlutterFlow/flutterflow-documentation: docs/integrations/maps/google-maps/google-maps-widget.md

---
title: Google Maps Widget
sidebar_position: 2
---

# Adding Google Maps widget

The Google Maps widget enables the integration of interactive maps into your app, offering users
valuable geographical insights. For instance, in a food delivery app, this widget could display the
locations of restaurants. It offers a range of customization options, allowing you to tailor the
display with various map types and markers to suit your specific needs.

:::danger[Feature Completion]
As we continuously enhance our platform, please note that while our integration is robust, it is not
yet feature-complete. We encourage you to review the available APIs and features detailed below to
ensure they meet your app development needs before integration.
:::

![google-maps-widget.png](..%2Fimgs%2Fgoogle-maps-widget.png)

:::info[Prerequisite]

> Ensure you have added the Google Map API keys before adding the Google Maps widget to your project

Reviewer: ```suggestion
Ensure you have added the [**Google Map API keys**](generate-maps-keys#get-api-keys) before adding the Google Maps widget to your project
```

:::

## Add Google Map widget

> - Open the Widget Palette and locate the **Google Map** widget under the **Base Elements** tab. You

Reviewer: An ordered list would be better.

  and
  enter the desired value. Note that a higher value will zoom in on the map while a lower value will
  zoom out.

<div class="video-container"><iframe src="https://www.loom.
com/embed/11c929da4a82492fb2d9e690779051d8?sid=477a3de8-3f0c-42e1-a8b6-7556c3e72e96" frameborder="0" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture; web-share" referrerpolicy="strict-origin-when-cross-origin" allowfullscreen></iframe></div>

<figure>

  <figcaption class="centered-caption"></figcaption>
</figure>
:::tip
If you don't see your current location while testing, make sure you have enabled location permission in your browser.

![location-browser.png](..%2Fimgs%2Flocation-browser.png)
:::

## Markers

A marker is an icon that appears over the map, indicating a location. To add markers:

- Select the **Google Map** widget, move to the **Properties Panel > Num Markers** and select
  whether you
  want to show **Single** or **Multiple** markers.

### Set Markers from Firebase

- Set the Marker Type to **Document** if the data is on Firestore Collection

> - Incase of Documents, create a collection and query it on any widget (must be a parent of

Reviewer: ```suggestion
- In case of Documents, create a collection and query it on any widget (must be a parent of
```

  GoogleMap) or page.

- In Marker Document, set the source of markers as shown in the following video.

<div class="video-container"><iframe src="https://www.loom.
com/embed/265b96cfd3554ce5b7516650b950b303?sid=83b3f4bd-8470-4307-9cbe-b742681c383f" frameborder="0" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture; web-share" referrerpolicy="strict-origin-when-cross-origin" allowfullscreen></iframe></div>

### Set Markers from List of LatLng

If you choose **LatLng**, you must provide a source that contains a list of locations as Data Type
(LatLng) (e.g., App State > [variable_name] (List of **LatLng**)).

<div class="video-container"><iframe src="https://www.loom.
com/embed/e0cf2e67ab0e45b7aedc3bb96b212d73?sid=38a4bdce-89f5-4278-aaa2-4ad8e859f925" frameborder="0" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture; web-share" referrerpolicy="strict-origin-when-cross-origin" allowfullscreen></iframe></div>

### Changing Marker Color

To change the marker color, move to the Properties Panel > Google Map > set the Marker Color
dropdown value to the color you like:

> ![marker-color.png](..%2Fimgs%2Fmarker-color.png)

Reviewer: ```suggestion
<p></p>
![marker-color.png](..%2Fimgs%2Fmarker-color.png)
```
